Sources/WebSocketKit/WebSocket.swift

```diff
@@ -272,22 +327,65 @@ public final class WebSocket: Sendable {
             } else {
                 self.close(code: .protocolError, promise: nil)
             }
+            
         case .text, .binary:
-            // create a new frame sequence or use existing
-            self.frameSequence.withLockedValue { currentFrameSequence in
-                var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
-                // append this frame and update the sequence
-                frameSequence.append(frame)
-                currentFrameSequence = frameSequence
+            // is compressed? and have pmce configured for the socket?
+            if frame.rsv1 , let pmce = pmce, pmce.enabled {
+               
+                do {
+                  
+                    if frame.maskKey != nil {
+                        
+                        let newFrame = try pmce.unmaskedDecompressedUnamsked(frame: frame)
+                        // create a new frame sequence or use existing
+                        self.frameSequence.withLockedValue { currentFrameSequence in
+                            var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
+                            // append this frame and update the sequence
+                            frameSequence.append(newFrame)
+                            currentFrameSequence = frameSequence
+                        }
+                    }
+                    else {
+                      
+                        let newFrame = try pmce.decompressed(frame)
+                        // create a new frame sequence or use existing
+                        self.frameSequence.withLockedValue { currentFrameSequence in
+                            var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
+                            // append this frame and update the sequence
+                            frameSequence.append(newFrame)
+                            currentFrameSequence = frameSequence
+                        }
+                    }
+                }
+                catch {
+                    print("websocket-kit: \(error)")
```

Comment:
Get rid of this print statement, and add some error handling. Of course, you can keep logging in here through `Logger`